### PR TITLE
fix: specific sheets to whole drive issue fixed

### LINKED ISSUE #21916: [Bug]: Spreadsheet dropdown does not show up all sheets when switched from specific sheets to all sheets
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Spreadsheet dropdown does not show all sheets, when google sheet datasource is edited from specific sheets to all sheets.

### Steps To Reproduce

1. Create google sheet datasource with selected sheets scope and select two sheets
2. Now edit the datasource and change scope from selected sheets to all sheets
3. Create a query for this datasource
4. Observe spreadsheet dropdown, it shows only 2 files, when it should show all files as scope is for all sheets

### Public Sample App

_No response_

### Environment

Deploy Preview

### Issue video log

_No response_

### Version

Cloud/ Self-Hosted - 1.8.4

### PR DESCRIPTION
**Pull request by sneha122**
## Description

This PR fixes the issue:
- Spreadsheet dropdown does not show all sheets, when google sheet datasource is edited from specific sheets to all sheets.
- If files are picked in the file picker, no error message is shown on datasource review page, but if files are not picked, error message is shown on review page.

Fixes #21916 

### How to test
First Issue:
- Create google sheet datasource with specific sheets modality
- Now create a query and check that only specific sheets are visible in spreadsheet dropdown
- Now edit the datasource and change scope from specific to all sheets
- Check the query and spreadsheet dropdown should show all sheets, as scope has changed

Second Issue:
- Create google sheet datasource, select files from file picker, no error message should be seen
- Edit the datasource, do not select files from file picker, error message should be seen.

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Manual

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/AuthenticationServiceCEImpl.java [hunk 1]
```diff
@@ -448,6 +446,15 @@ public Mono<OAuthResponseDTO> getAccessTokenFromCloud(String datasourceId, Strin
                                         projectID = authenticationResponse.getProjectID();
                                     }
                                 }
+
+                                // when authentication scope is other than specific sheets, we need to set authentication status as success
+                                // for specific sheets, it needs to remain in as in progress until files are selected
+                                if (oAuth2.getScope() != null && !oAuth2.getScope().contains(FILE_SPECIFIC_DRIVE_SCOPE)) {
```
Comment (sumitsum) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/AuthenticationServiceCEImpl.java:452]: Hi Sneha, can you please add a comment here mentioning that the success status is set by the client for specific sheets case. Also, please mention the client side file which contiains this change.

**Review by sumitsum — DISMISSED** [review]

**Comment by sumitsum** [comment]
server side changes LGTM.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/AuthenticationServiceCEImpl.java (hunk 1 above)
Comment (sneha122) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/AuthenticationServiceCEImpl.java:452]: got it will do that

**Review by Irongade — APPROVED** [review]

**Comment by sneha122** [comment]
> Deploy-Preview-URL: https://ce-21968.dp.appsmith.com

@prapullac

**Pull request merged**